homeassistant/components/tesla_fleet/button.py

```diff
@@ -0,0 +1,90 @@
+"""Button platform for Tesla Fleet integration."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from tesla_fleet_api.const import Scope
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TeslaFleetConfigEntry
+from .entity import TeslaFleetVehicleEntity
+from .helpers import handle_vehicle_command
+from .models import TeslaFleetVehicleData
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslaFleetButtonEntityDescription(ButtonEntityDescription):
+    """Describes a TeslaFleet Button entity."""
+
+    func: Callable[[TeslaFleetButtonEntity], Awaitable[Any]] | None = None
```

Comment:
why don't you make the api the parameter?

Reply:
Then he would need two parameters as he also needs coordinator data.
Easier to just pass the entity.